External/FEXCore/Source/Interface/Core/JIT/Arm64/MemoryOps.cpp

```diff
@@ -676,10 +676,7 @@ DEF_OP(LoadContextIndexed) {
     case 2:
     case 4:
     case 8: {
-      LoadConstant(ARMEmitter::Size::i64Bit, TMP1, Op->Stride);
-      mul(ARMEmitter::Size::i64Bit, TMP1, Index, TMP1);
-      add(ARMEmitter::Size::i64Bit, TMP1, STATE, TMP1.R());
-
+      add(ARMEmitter::Size::i64Bit, TMP1, STATE, Index, FEXCore::ARMEmitter::ShiftType::LSL, std::log2(Op->Stride));
```

Comment:
If I'm not mistaken, std::log2 is a *float* logarithm. While this works, it might incur bad code gen if the compiler isn't able to see through this. It also raises all sorts of "questionable semantic" warning bells. We know the stride is a power-of-two, we really just want the `__builtin_ctz` of it.

Should we add a FEX utility function that (in debug builds) asserts the input is a power-of-two and (regardless) returns the `__builtin_ctz`? That way the log2 becomes a single op in release builds even if this doesn't get constant foldd away (and I am unsure if it would be).

Reply:
`std::countr_zero` should be even better, actually.

Reply:
You're right. Created a helper and switched our usage over to it. Including a couple of lines that also used std::log2.
Added a unittest to ensure it returns expected data.